app/src/lib/shells/win32.ts

```diff
@@ -52,6 +57,20 @@ export async function getAvailableShells(): Promise<
     })
   }
 
+  const hyper = await readRegistryKeySafe(
+    'HKEY_CURRENT_USER\\Software\\Classes\\Directory\\Background\\shell\\Hyper\\command'
+  )
+  if (hyper.length > 0) {
+    const commandPieces = hyper[0].value.match(/(["'])(.*?)\1/)
```

Comment:
I'm not sure how I feel about this over using the install location `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Uninstall\hyper` and a for loop to get the install location. Could we do this?

```ts
const hyper = await readRegistryKeySafe(
  'HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Uninstall\hyper'
)
if (hyper.length > 0) {
  let path: string | null = null
  for (const item of hyper) {
    if (item.name === 'InstallLocation') {
      path = item.value
    }
  }
  if (installLocation) {
    shells.push({
      shell: Shell.Hyper,
      path,
    })
  }
}
```

Or even cheat a bit and do this?

```ts
const hyper = await readRegistryKeySafe(
  'HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Uninstall\hyper'
)
if (hyper.length > 0) {
  // this is a valid installation, and we know it's here, so don't bother reading the values
  shells.push({
    shell: Shell.Hyper,
    path: process.env.LocalAppData.concat('\\hyper\\Hyper.exe')
  })
}
```

Reply:
I agree that would be preferable. Unfortunately launching the `Hyper.exe` in the installation root doesn't allow specification of the start directory but the copy in the `app-x.x.x` does.

Reply:
@JordanMussi aha, I knew I missed some sort of detail. Yeah, using that registry key is better with the fallback to launch _something_...

Reply:
Do you think it would be better to check installation presence with the `HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Uninstall\hyper` key and then check the other key and use it if set?

Reply:
Nah, I assume the presence of this key is the same as it being installed.